src/wast-parser.cc

```diff
@@ -457,7 +474,7 @@ class ResolveFuncTypesExprVisitorDelegate : public ExprVisitor::DelegateNop {
       : module_(module), errors_(errors) {}
 
   void ResolveBlockDeclaration(const Location& loc, BlockDeclaration* decl) {
-    ResolveTypeNames(*module_, decl);
+    ResolveTypeNames(*module_, decl->sig, errors_);
```

Comment:
How is this different to the code in `src/wast-parser.cc` that does name resolving?

Reply:
I am not sure I understand this question. ResolveTypeNames is reworked to support error throwing.